pkg/virt-config/virt-config.go

```diff
@@ -208,11 +208,7 @@ func (c *ClusterConfig) SetVMISpecDefaultNetworkInterface(spec *v1.VirtualMachin
 		case v1.MasqueradeInterface:
 			spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
 		case v1.DeprecatedSlirpInterface:
-			if !c.IsSlirpInterfaceEnabled() {
-				return fmt.Errorf("Slirp interface is not enabled in kubevirt-config")
-			}
-			defaultIface := v1.DefaultSlirpNetworkInterface()
-			spec.Domain.Devices.Interfaces = []v1.Interface{*defaultIface}
+			return fmt.Errorf("slirp interface is deprecated as of v1.3")
```

Comment:
I'll try to summarize the current situation and options.

The problematic scenario is as follows:
- Cluster has Slirp defined in the KV CR: Slirp is the default binding in case no interfaces are defined.
- Cluster is upgraded to the version that no longer supports slirp. KV CR is not changed.
- An old VM is still running with no interfaces defined (i.e. it has Slirp binding implicitly).
- That old VM manifest can no longer be patched/updated because the VM admitter will block it on the Slirp setting existing on the KV CR as default and the absence of explicit interfaces in the VM manifest.
  - The running VM workload is not effected, it will continue running.

I suspect the scenario is very rare, because it assumes Slirp exists, VM has no interfaces defined and the default binding is set to be Slirp.

If an attempt is made to handle this case, we encounter problems with other scenarios.
Here are the cases that have been raised during conversations:
- Drop the validation from the VM admitter for update operations.
  - The problem here is that we cannot differentiate an old and new VM manifest. Someone may have deleted the interfaces and leave none, triggering the default validation and not failing the update.
- Add validation on the KV CR and check no one adds Slirp.
  - It will require a level based change check, which is usually not recomended. I.e. check the prev config vs the new config (no slirp before, now slirp was added).
  - If Slirp was there before, we still end up with it in the CR, which questions the need to have the prev check.
  - Having a regular validation on the KV to reject changes to it until the Slirp is removed is an option, but it is aggressive and may break the whole upgrade if external components try to set the KV CR with special settings (e.g. HCO).


To me, the limitation of the described scenario is rare enough to document and leave it as is.
The alternatives are more complex (logically and from the codebase perspective) and seem to present other limitations.

Reply:
ok, let's leave the PR as is. Just please update the docs with the fact slirp was deprecated and the consequences.